packages/notification-services-controller/src/NotificationServicesPushController/NotificationServicesPushController.ts

```diff
@@ -100,34 +99,25 @@ const metadata: StateMetadata<NotificationServicesPushControllerState> = {
   },
 };
 
-type ControllerConfig = {
+export type ControllerConfig = {
   /**
    * Config to turn on/off push notifications.
    * This is currently linked to MV3 builds on extension.
    */
   isPushEnabled: boolean;
 
   /**
-   * Must handle when a push notification is received.
-   * You must call `registration.showNotification` or equivalent to show the notification on web/mobile
+   * determine the config used for push notification services
    */
-  onPushNotificationReceived: (
-    notification: Types.INotification,
-  ) => void | Promise<void>;
+  platform: 'extension' | 'mobile';
 
   /**
-   * Must handle when a push notification is clicked.
-   * You must call `event.notification.close();` or equivalent for closing and opening notification in a new window.
+   * Push Service Interface
+   * - create reg token
+   * - delete reg token
+   * - subscribe to push notifications
    */
-  onPushNotificationClicked: (
-    event: NotificationEvent,
-    notification?: Types.INotification,
-  ) => void;
-
-  /**
-   * determine the config used for push notification services
-   */
-  platform: 'extension' | 'mobile';
+  pushService: PushService;
```

Comment:
The controller now must be initialised with this Push Service interface.
The `/web/push-helpers` can assist extension when injecting.
We will see if we can do the same for mobile.